src/PSADT/PSADT/Utilities/PSADT.Utilities.Class.IniUtilities.cs

```diff
@@ -34,5 +38,115 @@ public static void WriteSectionKeyValue(string section, string? key, string? val
         {
             Kernel32.WritePrivateProfileString(section, key, value, filepath);
         }
+
+        /// <summary>
+        /// Gets all key/value pairs in a section of an INI file.
+        /// </summary>
+        /// <param name="section">The section name</param>
+        /// <param name="filepath">Path to the INI file</param>
+        /// <returns>OrderedDictionary of key/value pairs in the section</returns>
+        public static OrderedDictionary? GetSection(string section, string filepath)
+        {
+            var sections = GetSectionNames(filepath);
+            if (!sections.Contains(section, StringComparer.OrdinalIgnoreCase))
+            {
+                throw new ArgumentException($"Section [{section}] was not found in the INI file.");
+            }
+
+            var buffer = new char[65536];
+
+            uint res;
+            try
+            {
+                res = Kernel32.GetPrivateProfileSection(section, buffer, filepath);
+            }
+            catch
+            {
+                return null;
+            }
+            if (res == 0)
+            {
+                return null;
+            }
+
+            var entries = new string(buffer, 0, (int)res).Split('\0');
+            var dictionary = new OrderedDictionary();
+
+            foreach (var entry in entries)
+            {
+                if (string.IsNullOrEmpty(entry)) continue;
+
+                var separatorIndex = entry.IndexOf('=');
+                if (separatorIndex <= 0) continue;
+
+                var key = entry.Substring(0, separatorIndex);
+                var value = entry.Substring(separatorIndex + 1);
+
+                if (dictionary.Contains(key))
+                {
+                    dictionary[key] = value;
+                }
+                else
+                {
+                    dictionary.Add(key, value);
+                }
+            }
+
+            return dictionary;
+        }
+
+        /// <summary>
+        /// Gets all section names from an INI file.
+        /// </summary>
+        /// <param name="filepath">Path to the INI file</param>
+        /// <returns>Array of section names</returns>
+        public static string[] GetSectionNames(string filepath)
+        {
+            var buffer = new char[65536];
+            var res = Kernel32.GetPrivateProfileSectionNames(buffer, filepath);
+            var names = new string(buffer, 0, (int)res).Split('\0').Where(name => !string.IsNullOrEmpty(name)).ToArray();
+
+            return names;
+        }
+
+        /// <summary>
+        /// Writes multiple key/value pairs to a section in an INI file.
+        /// </summary>
+        /// <param name="section">The section name</param>
+        /// <param name="keyValuePairs">Collection of key/value pairs to write</param>
+        /// <param name="filepath">Path to the INI file</param>
+        public static void WriteSection(string section, IDictionary? keyValuePairs, string filepath)
```

Comment:
As said above about why I understand the bulk of the code is using `OrderedDictionary`, I understand why this is `IDictionary` also for the same reason. Just be mindful that an `IDictionary` like a `Hashtable` or `OrderedDictionary` is effectively `Dictionary<object, object>`. As such, there's nothing stopping the caller from sending through complex values like hashtables, arrays, or other types that may not necessarily be representable as a string. It would be more technically correct to use `Dictionary<string, string>` for all of this code.